core/tests/widgets/test_webview.py

```diff
@@ -248,3 +248,53 @@ async def delayed_page_load():
 
     # The async handler was invoked
     on_result_handler.assert_called_once_with(42)
+
+
+async def test_get_cookies_async(widget):
+    """Cookies can be retrieved asynchronously from the WebView."""
+
+    # An async task that simulates retrieval of cookies after a delay
+    async def delayed_cookie_retrieval():
+        await asyncio.sleep(0.1)
+
+        # Simulate cookies being retrieved
+        cookies = {
```

Comment:
If this is a value that is being round tripped, it makes sense to define it once and then use that reference.